### PR TITLE
Bug 1846420: Re-vendor dependencies for manifest and code generation with go modules

### PR DESCRIPTION
**Pull request by kirankt**
New deep copy code generation
Adjust Makefile to use new vendoring
New CRDs generated by make manifest target

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ run: generate fmt vet
 
 # Install CRDs into a cluster
 install: manifests
-	kubectl apply -f vendor/github.com/openshift/cluster-api/config/crds/
+	kubectl apply -f vendor/github.com/openshift/machine-api-operator/install
```
Comment (sadasu) [inline Makefile:24]: What CRD(s) in the machine-api-operator does the CAPBM need to function?

Comment (kirankt) [reply Makefile:24]: The definitions of 'machine', 'machinesets', etc come from MAO (and formerly ClusterAPI). Thanks.

**Comment by mhrivnak** [comment]
> /assign @mhrivnak. PTAL at this PR, when you get a chance. A little context. The previous CAPI to MAO PR missed a few things, especially deepcopy and CRD generation. This PR remedies these overlooked parts. Many thanks.

Is there a gap in CI that explains why it wasn't caught the first time? Has additional testing of any kind been applied to this PR vs the prior one?

**Comment by stbenjam** [comment]
> > /assign @mhrivnak. PTAL at this PR, when you get a chance. A little context. The previous CAPI to MAO PR missed a few things, especially deepcopy and CRD generation. This PR remedies these overlooked parts. Many thanks.
> 
> Is there a gap in CI that explains why it wasn't caught the first time? Has additional testing of any kind been applied to this PR vs the prior one?

The previous PR passed... our e2e tests run a basic install with control plane and 2 workers, and a subset of the openshift conformance tests. What does this missing affect?

**Comment by stbenjam** [comment]
Also: this needs a BZ so we can backport it to 4.5, the previous PR was against 4.5.

**Comment by kirankt** [comment]
> > > /assign @mhrivnak. PTAL at this PR, when you get a chance. A little context. The previous CAPI to MAO PR missed a few things, especially deepcopy and CRD generation. This PR remedies these overlooked parts. Many thanks.
> > 
> > 
> > Is there a gap in CI that explains why it wasn't caught the first time? Has additional testing of any kind been applied to this PR vs the prior one?
> 
> The previous PR passed... our e2e tests run a basic install with control plane and 2 workers, and a subset of the openshift conformance tests. What does this missing affect?

Basically we're not checking a few makefile targets (manifests, CRD and code generation, etc). The e2e worked because the binary built fine and worked correctly.

**Comment by dhellmann** [comment]
> > > > /assign @mhrivnak. PTAL at this PR, when you get a chance. A little context. The previous CAPI to MAO PR missed a few things, especially deepcopy and CRD generation. This PR remedies these overlooked parts. Many thanks.
> > > 
> > > 
> > > Is there a gap in CI that explains why it wasn't caught the first time? Has additional testing of any kind been applied to this PR vs the prior one?
> > 
> > 
> > The previous PR passed... our e2e tests run a basic install with control plane and 2 workers, and a subset of the openshift conformance tests. What does this missing affect?
> 
> Basically we're not checking a few makefile targets (manifests, CRD and code generation, etc). The e2e worked because the binary built fine and worked correctly.

Do we have a ticket for adding test jobs for those missing checks?

**Comment by kirankt** [comment]
> > > > > /assign @mhrivnak. PTAL at this PR, when you get a chance. A little context. The previous CAPI to MAO PR missed a few things, especially deepcopy and CRD generation. This PR remedies these overlooked parts. Many thanks.
> > > > 
> > > > 
> > > > Is there a gap in CI that explains why it wasn't caught the first time? Has additional testing of any kind been applied to this PR vs the prior one?
> > > 
> > > 
> > > The previous PR passed... our e2e tests run a basic install with control plane and 2 workers, and a subset of the openshift conformance tests. What does this missing affect?
> > 
> > 
> > Basically we're not checking a few makefile targets (manifests, CRD and code generation, etc). The e2e worked because the binary built fine and worked correctly.
> 
> Do we have a ticket for adding test jobs for those missing checks?

We don't. I'll add one right away.

**Comment by kirankt** [comment]
/retitle Bug 1846420: Re-vendor dependencies for manifest and code generation with go modules

**Comment by kirankt** [comment]
Can a I get a LGTM, please? I have a PR open to add CI jobs to catch these in the future:
https://github.com/openshift/release/pull/9713. This current PR needs to merge before the CI PR to pass.

**Pull request merged**